baseline-refaster-rules/src/main/java/com/palantir/baseline/refaster/AssertjMapDoesNotContainKey.java

```diff
@@ -0,0 +1,50 @@
+/*
+ * (c) Copyright 2019 Palantir Technologies Inc. All rights reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package com.palantir.baseline.refaster;
+
+import static org.assertj.core.api.Assertions.assertThat;
+
+import com.google.errorprone.refaster.ImportPolicy;
+import com.google.errorprone.refaster.annotation.AfterTemplate;
+import com.google.errorprone.refaster.annotation.BeforeTemplate;
+import com.google.errorprone.refaster.annotation.UseImportPolicy;
+import java.util.Map;
+
+public final class AssertjMapDoesNotContainKey<K, V> {
+
+    @BeforeTemplate
+    void before1(Map<K, V> things, K key) {
+        assertThat(things.containsKey(key)).isFalse();
+    }
+
+    @BeforeTemplate
+    void before2(Map<K, V> things, K key) {
+        assertThat(things.get(key)).isNull();
```

Comment:
I'm not sure we necessarily want to refactor this one to `assertThat(things).doesNotContainKey(key)` as while this is true for some implementations of map (e.g. `ConcurrentHashMap`, `ImmutableMap`), but not necessarily for others such as `HashMap` or `TreeMap` where a key may be mapped to `null` (i.e. `things.get(key) == null && things.containsKey(key) == true`)

Reply:
To be fair, this did correctly refactor several places in tritium where we can properly use `doesNotContainKey` in https://github.com/palantir/tritium/pull/454

Reply:
I've found one place in practice where it broke a test, but in all other cases it produced better asserts. When this is hit, we either want to refactor to:
`assertThat(map).doesNotContainKey(key)`
or
`assertThat(map).containsEntry(key, null)`

The two are semantically different, and it's uncommon that test authors actually mean to test that either the value is null or the map does not contain a key.